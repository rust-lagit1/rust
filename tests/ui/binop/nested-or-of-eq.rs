fn main() {
    let x = 1;
    if x == 1 || 2 || 3 {
        //~^ ERROR mismatched types
        //~| ERROR mismatched types
        println!("Was 1 or 2 or 3");
    }

    let x = 1.0;
    if x == 1.0 || 2.0 || 3.0 {
        //~^ ERROR mismatched types
        //~| ERROR mismatched types
        println!("Was 1.0 or 2.0 or 3.0");
    }
}
