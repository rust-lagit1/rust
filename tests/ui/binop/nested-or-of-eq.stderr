error[E0308]: mismatched types
  --> $DIR/nested-or-of-eq.rs:3:18
   |
LL |     if x == 1 || 2 || 3 {
   |        ------    ^ expected `bool`, found integer
   |        |
   |        expected because this is `bool`
   |
help: use `matches!()` to match against multiple values
   |
LL |     if matches!(x, 1 | 2 | 3) {
   |        +++++++++ ~   ~   ~  +

error[E0308]: mismatched types
  --> $DIR/nested-or-of-eq.rs:3:23
   |
LL |     if x == 1 || 2 || 3 {
   |        -----------    ^ expected `bool`, found integer
   |        |
   |        expected because this is `bool`

error[E0308]: mismatched types
  --> $DIR/nested-or-of-eq.rs:10:20
   |
LL |     if x == 1.0 || 2.0 || 3.0 {
   |        --------    ^^^ expected `bool`, found floating-point number
   |        |
   |        expected because this is `bool`
   |
help: use `x == ...` to match against several different values
   |
LL |     if x == 1.0 || x == 2.0 || x == 3.0 {
   |                    ++++        ++++

error[E0308]: mismatched types
  --> $DIR/nested-or-of-eq.rs:10:27
   |
LL |     if x == 1.0 || 2.0 || 3.0 {
   |        ---------------    ^^^ expected `bool`, found floating-point number
   |        |
   |        expected because this is `bool`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
